### ISSUE TITLE
 java.lang.IllegalArgumentException: no suitable constructors found for ZonedDateTime

### ISSUE THREAD

**Issue by mallim** [orika-mapper/orika#96]
Version used is 1.4.6 running in JDK 1.8

The following exception is happening:

```
Caused by: java.lang.IllegalArgumentException: no suitable constructors found for ZonedDateTime
        at ma.glasnost.orika.impl.generator.ObjectFactoryGenerator.addSourceClassConstructor(ObjectFactoryGenerator.java:165) ~[orika-core-1.4.6.jar:na]
        at ma.glasnost.orika.impl.generator.ObjectFactoryGenerator.addCreateMethod(ObjectFactoryGenerator.java:124) ~[orika-core-1.4.6.jar:na]
        at ma.glasnost.orika.impl.generator.ObjectFactoryGenerator.build(ObjectFactoryGenerator.java:95) ~[orika-core-1.4.6.jar:na]
```

Thanks

**Comment by elaatifi** [comment]
ZonedDateTime can not be cloned by Orika due the lack of constructor...
Usually for this "immutable" value object the best thing to do is to copy by reference (by registering CopyByReferenceConverter with ZonedDateTime.class as argument) but you can also use a custom converter to if the type does not match.

**Comment by lbehnke** [comment]
I would be great if these immutable objects (LocalDateTime, ZonedDateTime, ...) are automatically recognized as such and just passed through during the mapping. As they are part of the JDK, I do not see any reason why application developers must deal with the problem.

**Comment by elaatifi** [comment]
_We can verify in the runtime the existence of these classes (Class.forName) before registering the CopyByReferenceConverter_

**Comment by brabenetz** [comment]
Duplicate with #170

**Comment by andreabertagnolli** [comment]
Fixed by #178
